PublicOnboardingDemo/clients/LandingPage/controllers/client/onStart.js

```diff
@@ -0,0 +1,32 @@
+    
+    var isPublic = client.isPublic();
+    
+    var intro = client.getWidget("WelcomeLine");
+    intro.text = isPublic ? "Welcome to VANTIQ! You are currently in Public Mode." : "Welcome to VANTIQ! You are currently in Private Mode.";
+    
+    var uuid = client.getUsername();
+    var username = client.getUserRecord();
+    var uuid_widget = client.getWidget("UUID");
+    var username_widget = client.getWidget("Username");
+    var pref_username = "";
+    
+    uuid_widget.text = "Your Unique User ID is: " + uuid;
+    
+    // at this point username returns null in some cases
+    // the following function checks every tenth of a second
+    // to try and request the username data - and also sets and displays
+    // the username of the current user
+    function getPrefUsername(){
```

Comment:
I would rename this `setPrefUsername` or `updatePrefUsername`, since the main intention is to update the text in the UI with the correct preferred username whenever it becomes available.